analyses/infercnv-consensus-cell-type/exploratory-notebooks/SCPCP000004/01_nb-consensus-cell-types.Rmd

```diff
@@ -0,0 +1,238 @@
+We can also determine whether any samples can be run with an internal (non-pooled) reference as a control.
+
+## Setup
+
+```{r}
+options(readr.show_col_types = FALSE)
+library(ggplot2)
+theme_set(theme_bw())
+```
+
+### Paths
+
+#### Base directories
+
+
+```{r base paths}
+# The base path for the OpenScPCA repository, found by its (hidden) .git directory
+repository_base <- rprojroot::find_root(rprojroot::is_git_root)
+
+# The current data directory, found within the repository base directory
+data_dir <- file.path(repository_base, "data", "current", "SCPCP000004")
+
+# The current results directory, found within the repository base directory
+result_dir <- file.path(repository_base, "data", "current", "results")
+
+# Path to cell-type-consensus results
+consensus_dir <- file.path(result_dir, "cell-type-consensus", "SCPCP000004")
+
+# Path to reference cell type groupings
+reference_dir <- file.path(repository_base, "analyses", "infercnv-consensus-cell-type", "references")
+```
+
+#### Input and output files
+
+```{r paths}
+# Input consensus cell type files
+celltype_files <- list.files(
+  path = consensus_dir,
+  pattern = "_processed_consensus-cell-types\\.tsv\\.gz$",
+  recursive = TRUE,
+  full.names = TRUE
+) |>
+  # set names as library id
+  purrr::set_names(
+    \(x) {
+      stringr::str_split_i(basename(x), "_", 1)
+    }
+  )
+
+# project sample metadata file
+metadata_file <- file.path(data_dir, "single_cell_metadata.tsv")
+
+# reference cell type groups
+reference_file <- file.path(reference_dir, "reference-cell-groups.tsv")
+```
+
+
+## Analysis
+
+```{r}
+# determine PDX libraries to exclude
+tumor_library_ids <- readr::read_tsv(metadata_file) |>
+  dplyr::filter(!is_xenograft) |>
+  dplyr::pull(scpca_library_id)
+
+# get cell groups
+cell_group_df <- readr::read_tsv(reference_file)
+
+# Prepare data frame of all cell types
+celltype_df <- celltype_files |>
+  # consider only non-PDX
+  purrr::keep_at(tumor_library_ids) |>
+  purrr::map(readr::read_tsv) |>
+  purrr::list_rbind() |>
+  dplyr::select(library_id, barcodes, consensus_annotation) |>
+  dplyr::left_join(cell_group_df, by = "consensus_annotation")
+```
+
+We'll first exclude any libraries that have fewer than 1000 cells from consideration in the reference.
+
+```{r}
+discard_libraries <- celltype_df |>
+  dplyr::count(library_id) |>
+  dplyr::filter(n < 1000) |>
+  dplyr::pull(library_id)
+
+discard_libraries
+```
+
+```{r}
+celltype_df <- celltype_df |>
+  dplyr::filter(!(library_id %in% discard_libraries))
+```
+
+These cell groupings are present in the neuroblastoma samples we're considering:
+
+```{r}
+unique(celltype_df$reference_cell_group)
+```
+
+We should remove any cell groups which might correspond to tumor cells, including `stromal`, `neuronal`, and `stem cell`.
+The `NA` values here correspond to `Unknown` cells, which we'll also remove.
+This means we'll keep groups `endothelial`, `epithelial`, and `immune`.
```

Comment:
```suggestion
This means we'll keep groups `endothelial`, `epithelial`, `adipocyte`, and `immune`.
```
I don't know that we have a specific reason to exclude `adipocyte` here, so I would keep them in the "all normal" reference.